### PR TITLE
Fix type of empty `Index` and raise warning in `Series` constructor

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
Fixes: #14091 
This PR fixes empty inputs dtype in `Index` to default to `str` instead of `float64`. Another change is there is a deprecation warning for `Series` constructor to match pandas.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/tests/test_dataframe.py [hunk 1]
```diff
@@ -2001,7 +2002,7 @@ def test_series_shape():
 
 def test_series_shape_empty():
     ps = pd.Series(dtype="float64")
-    cs = cudf.Series([])
+    cs = _create_cudf_series([])
```
Comment (bdice) [inline python/cudf/cudf/tests/test_dataframe.py:2005]: Rather than call this helper function to make cudf use a float64, shouldn't we alter this test to call `pd.Series([])`?

File: python/cudf/cudf/tests/test_dataframe.py [hunk 2]
```diff
@@ -2840,7 +2841,7 @@ def test_series_all_null(num_elements, null_type):
 @pytest.mark.parametrize("num_elements", [0, 2, 10, 100])
 def test_series_all_valid_nan(num_elements):
     data = [np.nan] * num_elements
-    sr = cudf.Series(data, nan_as_null=False)
+    sr = _create_cudf_series(data, nan_as_null=False)
```
Comment (bdice) [inline python/cudf/cudf/tests/test_dataframe.py:2844]: Why is this change needed?

For `num_elements == 0`, the null count should still be zero even if it interprets the dtype as string/object (so the test is still valid, though it might warn?).

For `num_elements > 0`, the data should be non-empty and thus be correctly interpreted as a floating type.

File: python/cudf/cudf/tests/test_dataframe.py [hunk 3]
```diff
@@ -4073,28 +4074,28 @@ def test_empty_dataframe_describe():
 
 
 def test_as_column_types():
-    col = column.as_column(cudf.Series([]))
+    col = column.as_column(_create_cudf_series([]))
```
Comment (bdice) [inline python/cudf/cudf/tests/test_dataframe.py:4077]: Same here and below -- I'm not sure if I understand why these helper functions are needed. It seems like the test construction might need subtle changes but I don't think this casting helper function is the right answer. It's not announcing what it is actually doing -- the differences between `cudf.Series` and `_create_cudf_series` should be in the name, if this is needed (like `_create_cudf_series_float64_default`).

**Review by bdice — CHANGES_REQUESTED** [review]

File: python/cudf/cudf/tests/test_dataframe.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/tests/test_dataframe.py]: Changed it to `Series([], dtype="float64")` and other places below. But retained the helper function calls in few places which can be used generically.

File: python/cudf/cudf/tests/test_dataframe.py (hunk 3 above)
Comment (galipremsagar) [reply python/cudf/cudf/tests/test_dataframe.py]: Agreed, reverted the change here and passed explicit dtypes. Renamed the helper to `_create_cudf_series_float64_default` and `_create_pandas_series_float64_default`. These will be removed in `pandas-2.0` upgrade.

**Review by bdice — APPROVED** [review]
Approving -- but let's make sure to remove these testing helper functions early in the pandas 2.0 migration.
